extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,173 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/** @const {!Object<!Object<string, !Array<function>>>} */
+
+const productImagesValidation = {
+  'url': [validateRequired, validateURLs],
+  'alt': [validateRequired, validateString],
+};
+
+const aggregateRatingValidation = {
+  'ratingValue': [validateRequired, validateNumber],
+  'reviewCount': [validateRequired, validateNumber],
+  'reviewUrl': [validateRequired, validateURLs],
+};
+
+export const productValidationConfig = {
+  /* Required Attrs */
+  'productUrl': [validateRequired, validateURLs],
+  'productId': [validateRequired, validateString],
+  'productTitle': [validateRequired, validateString],
+  'productBrand': [validateRequired, validateString],
+  'productPrice': [validateRequired, validateNumber],
+  'productImages': [
+    validateRequired,
+    createValidateConfigArray(productImagesValidation),
+  ],
+  'productPriceCurrency': [validateRequired, validateString],
+  /* Optional Attrs */
+  'aggregateRating': [createValidateConfigObject(aggregateRatingValidation)],
```

Comment:
The nomenclature was decided here https://github.com/ampproject/amphtml/pull/37474#discussion_r812244168
these functions are only the initial config, but we also have to take into consideration that the config / validation will be changed and scale over time. Rather than having specific functions for each config object, we have two generic functions `createValidateConfigObject ` and `createValidateConfigArray ` that can be re-used for any validation object (with a customized validationObject which defines the validation functions that will be called) as the code scales and more config objects are added in V2.